src/honeybadger.cr

```diff
@@ -3,30 +3,70 @@ require "./honeybadger/*"
 module Honeybadger
   VERSION = "0.1.0"
 
-  @@api_key = ""
-  @@report_data = true
+  class Configuration
+    # A Honeybadger API key.
+    property api_key = ""
+
+    # The API endpoint for sending Honeybadger payloads.
+    property endpoint : Path = Path["https://api.honeybadger.io"]
+
+    # The project git revision. Evaluated at compile time.
+    getter git_revision : String = {{ run("./run_macros/git_revision.cr").stringify }}.strip
```

Comment:
Let's rename this option to just "revision" (which is what our other client libraries call it). It should still default to the git revision like it currently does. In some cases people aren't using git, in which case revision can be a hash from a different scm system of a version number.

Reply:
Makes sense, updated!